charts/castai-hosted-model/child-charts/vllm/templates/deployment.yaml

```diff
@@ -130,6 +148,56 @@ spec:
             - name: llm-cache
               mountPath: /cache
           {{ end }}
+            - name: models
+              mountPath: /models
+      initContainers:
+{{- if .Values.loraAdapter.name }}
+        - name: model-downloader
+          image: "{{ .Values.modelDownloader.image.repository }}:{{ .Values.modelDownloader.image.tag }}"
+          imagePullPolicy: IfNotPresent
+          args:
+            - download
+          env:
+            - name: STORAGE_TYPE
```

Comment:
nit: Passing the storage related env vars and volume mounts is basically the same as in S3 proxy. For now the code duplicity is not a big deal, but once we will start adding other providers this will bloat. Would be a good idea to refactor it a bit, so we can reuse that part.

Reply:
Updated.